psydac/linalg/basic.py

```diff
@@ -900,10 +928,39 @@ def dtype(self):
     @property
     def linop(self):
         return self._A
+    
+    @linop.setter
+    def linop(self, a):
+        assert isinstance(a, LinearOperator)
+        assert a.domain is self.domain
+        assert a.codomain is self.codomain
+        self._A = a
```

Comment:
I agree that any use of this setter should be made with great care, and if possible it should be avoided. Sometimes, however, can occur an operator of the form `A - dt*B` where the time step `dt` changes for example. In this case I would prefer the setter instead of creating a new object and passing all parameters again. Or is there another possibility?

From a design point of view, the setter allows for greater flexibility, which usually comes with greater danger of erroneous use. At some point the developer must know what he is doing.

Reply:
Indeed it should be possible to modify `dt` directly in the `ScaledLinearOperator` object `dt * B`.

If I remember correctly @jowezarek has added this feature to a private fork of Psydac, and uses it for variable time stepping. Would this work for you?

Reply:
Update: For more complicated expressions replacing a scalar parameter could be practically impossible (especially if we take powers of such a parameter...). The setter for `linop` is needed in such a cases. Its docstring should point out that its usage should be very careful.

Reply:
Docstring added in 54f4671